### PR TITLE
feat: delivery domain completion + media buy test coverage (v3.6)

### PR DESCRIPTION
**Pull request by KonstantinMirin**
## Summary

Completes the Delivery entity vertical slice and expands Media Buy test coverage
for v3.6. Extracts delivery schemas into a standalone package, creates the
DeliveryRepository, fills 127+ behavioral obligation test stubs, and replaces
dict-based v2 compat with model-level serialization.

Merged with latest main (proper two-parent merge). All test suites pass locally.

## Key Changes

| Commit | Type | Description |
|--------|------|-------------|
| `35166f48` | fix | **CPC clicks enum comparison bug** — `PricingModel.cpc` vs string `"cpc"` never matched; click derivation was silently disabled |
| `35166f48` | fix | Remove dead top-level `pricing_option_id` lookup (not in AdCP spec) |
| `22dfdcfa` | refactor | Remove dead `get_principal_id_from_context` function (replaced by ResolvedIdentity) |
| `1ac07a4e` | test | 4 integration tests for webhook_delivery_service.py edge cases |
| `d7d8c36f` | fix | Restore 110 obligation tests lost in auto-merge, fix `get_product_catalog` patch target |
| `cee9eb3d` | chore | Proper merge with upstream main (two-parent, resolves GH conflict view) |
| `e0aa992a` | fix | Add missing migration `886966ee9a9d` + remove stale `get_creative_delivery` tool registration |

### Bug found: CPC clicks derivation was silently broken

Line 386 compared `pricing_option.pricing_model` (raw string `"cpc"` from DB) against
`PricingModel.cpc` (Python enum). Python enums don't compare equal to their `.value` —
so the CPC click derivation formula `clicks = floor(spend / cpc_rate)` **never executed**.
Fixed by comparing against `PricingModel.cpc.value`.

### Delivery scoped coverage: 97.9%

```
media_buy_delivery.py          95.4%  (14 uncovered)
webhook_delivery.py           100.0%  (0 uncovered)
webhook_delivery_service.py    98.7%  (3 uncovered)
delivery.py                   100.0%  (0 uncovered)
```

## Local test results (all passing)

| Suite | Result |
|-------|--------|
| Unit | 3398 passed, 3 xfailed |
| Integration | 774 passed, 26 skipped, 4 xfailed |
| Integration V2 | 172 passed, 10 skipped, 1 xfailed |
| E2E | 81 passed, 25 skipped |
| UI | OK |

## Prior Changes

| Commit | Type | Description |
|--------|------|-------------|
| `c7c2727c` | feat | DeliveryRepository for webhook delivery tables (+701 integration test lines) |
| `479f3ca8` | refactor | Extract delivery schemas → `src/core/schemas/delivery.py` package |
| `95d05cf3` | refactor | Tighten delivery schemas against adcp spec (FIXMEs for naming mismatches) |
| `0deff46f` | feat | 17 UC-004 delivery obligation tests (metrics, webhooks, retry/error) |
| `e9404090` | refactor | Model-level v2 serialization replacing dict-based `add_v2_compat` |
| `09798d8f` | feat | 54 UC-003 update media buy obligation tests |
| `a9922818` | feat | 56 UC-002 create media buy obligation tests |
| `15213478` | chore | Consolidate obligation allowlist |

### Delivery domain extraction
- **DeliveryRepository** (`src/core/database/repositories/delivery.py`):
  webhook subscription and delivery log queries with tenant-scoped access
- **Schema package** (`src/core/schemas/delivery.py`): extracted from monolithic
  `schemas.py`; aligned against adcp spec — added `flight_ended`/`goal_met` to
  `DeliveryStatus`, documented naming mismatches (`video_completions` vs spec
  `completed_views`), added FIXMEs for webhook metadata placement
- Integration tests for DeliveryRepository (701 lines)

### Integration test harness + factory_boy
- **Test harness** (`tests/harness/`): `IntegrationEnv` base with factory session binding;
  domain envs (`DeliveryPollEnv`, `WebhookEnv`, `CircuitBreakerEnv`) mock only external
  services, real DB for everything else
- **Factory_boy factories** (`tests/factories/`): `TenantFactory`, `PrincipalFactory`,
  `ProductFactory`, `PricingOptionFactory`, `MediaBuyFactory`, `MediaPackageFactory`,
  `PushNotificationConfigFactory` with SubFactory cascading

### Obligation test coverage (127+ tests)
- **UC-004** (17 tests): delivery metrics, webhook push reporting, retry/error handling
- **UC-002** (56 tests): create media buy — approval workflows, timing, creatives,
  proposals, extensions, postconditions, upgrade paths
- **UC-003** (54 tests): update media buy — budget ops, creative updates, pause/resume,
  targeting overlay, manual approval, extensions A–O

## Related
- Part of the v3.6 upgrade tracked in #1043
- References #1038 (GAM sync — some UC-003 obligations deferred)
- References #1075 (idempotency_key — some UC-003 obligations deferred)

### CODE REVIEW

**Comment by KonstantinMirin** [comment]
## Addressed Review Comments (commit 85244d8b)

Thanks for the thorough review @ChrisHuie! All 5 actionable items addressed:

### 1. Vacuous-pass tests (`except Exception` + `result = None`)
**Fixed.** 7 tests now catch `AdCPValidationError` specifically and assert the targeted validation step didn't fail. Other exceptions pass through (downstream failures unrelated to the validation being tested are fine). Lines 358, 436, 1151, 1180, 1223, 1466, 2200 — all no longer silently pass when impl raises.

### 2. `_is_circuit_breaker_open` private dict access
**Fixed.** Added `has_open_circuit_breaker(tenant_id: str) -> bool` public method to `WebhookDeliveryService`. `_is_circuit_breaker_open()` now delegates to it. Note: `get_circuit_breaker_state()` wasn't a drop-in — it checks one endpoint, we needed tenant-wide scan.

### 3. `protocol_webhook_service.py` missing URL validation
**Fixed.** Added `WebhookURLValidator.validate_webhook_url()` call in `send_notification()` before localhost normalization, matching all other webhook senders.

### 4. Misplaced `DailyBreakdown` fields
**Removed.** The 5 response-level webhook metadata fields (`notification_type`, `partial_data`, `unavailable_count`, `sequence_number`, `next_expected_at`) were confirmed dead code — never populated on any `DailyBreakdown` instance. They belong on `GetMediaBuyDeliveryResponse` (already inherited from adcp library). The broader FIXMEs (`video_completions` rename etc.) are tracked separately.

### 5. `_get_pricing_options` type annotations
**Fixed.** `list[Any | None]` → `list[str]`, `tenant_id: str | None = None` → `tenant_id: str` (required). Sole caller always provides both as strings.

### 6. `get_db_session` allowlist expansion
Filed `salesagent-r15q` to investigate all allowlist entries and scope the repository refactoring. The current entry for `_get_pricing_options` is a pre-existing violation from the PricingOption query fix — moving to a repository requires updating ~15 test mock sites and is out of scope for this PR.

---

`make quality`: 3398 passed, all structural guards green.

File: src/core/schemas/delivery.py [hunk 1]
```diff
@@ -0,0 +1,510 @@
...
+        ge=1,
+    )
+    next_expected_at: str | None = Field(
+        None,
+        description="MISPLACED: response-level webhook field, not per-day. See GetMediaBuyDeliveryResponse.",
+    )
+
+    date: str = Field(description="Date (YYYY-MM-DD)", pattern=r"^\d{4}-\d{2}-\d{2}$")
+    impressions: float = Field(ge=0, description="Daily impressions")
+    spend: float = Field(ge=0, description="Daily spend")
```
[the remark this answers is not in the document]
Comment (KonstantinMirin) [reply src/core/schemas/delivery.py]: Removed in \`85244d8b\`. Confirmed these 5 fields (\`notification_type\`, \`partial_data\`, \`unavailable_count\`, \`sequence_number\`, \`next_expected_at\`) were never populated on any \`DailyBreakdown\` instance — they belong on \`GetMediaBuyDeliveryResponse\` (response-level metadata), not on the per-day model. No consumers found.
